### PR TITLE
HV-1408 Add small usage tests for each HV annotation

### PR DESCRIPTION
**Pull request by marko-bekhta**
- https://hibernate.atlassian.net/browse/HV-1408

### CODE REVIEW

File: engine/src/test/java/org/hibernate/validator/test/constraints/annotations/AbstractConstrainedTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package org.hibernate.validator.test.constraints.annotations;
```
Comment (gunnarmorling) [inline engine/src/test/java/org/hibernate/validator/test/constraints/annotations/AbstractConstrainedTest.java]: The files are missing the right license hearder. You can add them by running `mvn license:format -pl engine`.

File: engine/src/test/java/org/hibernate/validator/test/constraints/annotations/hv/EmailConstrainedTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.hibernate.validator.constraints.Email;
+import org.hibernate.validator.test.constraints.annotations.AbstractConstrainedTest;
+
+import org.testng.annotations.Test;
+
+/**
+ * @author Marko Bekhta
+ */
+@SuppressWarnings("depricated")
```
Comment (gunnarmorling) [inline engine/src/test/java/org/hibernate/validator/test/constraints/annotations/hv/EmailConstrainedTest.java]: s/deprecited/deprecated/

**Comment by marko-bekhta** [comment]
Thanks @gunnarmorling, just got to this. It's strange that headers are missing as I remember adding them (at least I think I do:) ). 
As for the removed tests - I actually just moved them between packages. As for the tests themselves - I tried to make them as simple as possible to just test positive/negative scenario to make sure that all implemented validators are picked up. 

I was also thinking to maybe rewrite the country specific constraints tests - to have a test for the validator and in the tests of this new package just do a simple verification that annotations work. WDYT?

**Comment by gsmet** [comment]
> I was also thinking to maybe rewrite the country specific constraints tests - to have a test for the validator and in the tests of this new package just do a simple verification that annotations work. WDYT?

I don't think it's worth it.

Btw, I removed the package-info file as I don't think it's time well spent to add them for the tests. Let's only add them for the main packages.

Merged. Thanks!

**Pull request closed without merging**